feat: add C implementation for `dist/binomial/entropy`

sethdivyansh opened the pull request:
Resolves #3466

## Description

- adds C implementation for @stdlib/stats/base/dists/binomial/entropy along with relevant tests, benchmarks and examples

This pull request:

-  progresses: #3466 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

sethdivyansh committed:
feat: add C implementation for dist/binomial/entropy

sethdivyansh committed:
style: remove unnecessary blank line in C example for binomial entropy

sethdivyansh committed:
fix: correct indexing in benchmark tests for binomial entropy

sethdivyansh committed:
fix: random number generation in benchmark tests for binomial entropy

sethdivyansh committed:
fix: add missing dependency for ceil function in binomial entropy manifest

Planeshifter commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

Planeshifter committed:
Merge branch 'develop' into binomial-entropy

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter reviewed (changes requested):
Thanks for opening this PR! 

It should be refactored such that `n` is typed as `int32_t`, see other merged binomial C implementations, e.g. https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/base/dists/binomial/mean.

Planeshifter committed:
chore: minor clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/binomial/entropy/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,153 @@
+
+	for ( i = 0; i < 100; i++ ) {
+		n[ i ] = random_int( 0, 100 );
+		p[ i ] = random_uniform( 0.0, 1.0);
```
```suggestion
		p[ i ] = random_uniform( 0.0, 1.0 );
```

Planeshifter committed:
style: add missing space

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
Merge branch 'develop' into binomial-entropy

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter reviewed (approved):
LGTM after clean-up. Thanks, @sethdivyansh!
